boskos/gcp/gke.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"google.golang.org/api/container/v1"
+)
+
+const (
+	operationTimeout = 15 * time.Minute
+	defaultSleepTime = 10 * time.Second
+)
+
+type clusterConfig struct {
+	MachineType           string `json:"machinetype,omitempty"`
+	NumNodes              int64  `json:"numnodes,omitempty"`
+	Version               string `json:"version,omitempty"`
+	Zone                  string `json:"zone,ompitempty"`
+	EnableKubernetesAlpha bool   `json:"enablekubernetesalpha"`
+}
+
+type containerEngine struct {
+	service *container.Service
+}
+
+func findVersionMatch(version string, supportedVersion []string) (string, error) {
+	for _, v := range supportedVersion {
+		if strings.HasPrefix(v, version) {
+			return v, nil
+		}
+	}
+	return "", nil
+}
+
+func (cc *containerEngine) waitForOperation(ctx context.Context, op *container.Operation, project, zone string) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-time.After(defaultSleepTime):
+			newOp, err := cc.service.Projects.Zones.Operations.Get(project, zone, op.Name).Context(ctx).Do()
+			if err != nil {
+				return err
+			}
+			switch newOp.Status {
+			case "DONE":
```

Comment:
Are these string standardized? Maybe define they as enum instead of hardcoding?

Reply:
are there no other values? default?

Reply:
added logging